### ISSUE TITLE
Make setup.py the source of truth

### ISSUE THREAD

**Issue by jsignell** [holoviz/holoviz#154]
Move everything but license from setup.cfg to setup.py

**Comment by jbednar** [comment]
We decided this in a PyViz team meeting just now. For background, we'd ideally like there to be a single source of truth for the project-specific declarations (dependencies, etc.) that we need, consistently across our projects.  The current pyviz configuration is the result of attempting to use `setup.cfg` to be this source of truth, motivated by (a) setup.cfg currently being required by setuptools for wheels to include the license files they need, and (b) setup.cfg being declarative and thus reasonable for other tools to parse and extract data.  

After discussion, however, we decided to use setup.py for this purpose instead, as currently done in other pyviz projects, because (a) both setup.py and setup.cfg are currently required (e.g. setup.py being needed for `pip install -e .` or `python setup.py develop` to work) so we can choose either one we prefer, (b) some extreme measures that weren't quite declarative anyway were required to get setup.cfg to work (e.g. `version = attr: param.version.get_setupcfg_version` (which calls a function defined in param.version, which in turn depends on get_setupcfg_version parsing setup.cfg to pull out the git SHA left there by git archive...),  (c) setup.cfg only works with current pip versions that support pyproject.toml-declared build-time dependencies (so that param.version is available), (d) setup.py is just Python code and thus understandable by everyone working on these projects, while setup.cfg is mysterious and harder for us to figure out, and (c) setup.py is the more typical and expected approach for Python projects, which isn't definitive but is a vote in its favor when the balance is otherwise close.  

Basically, setup.cfg doesn't really solve all our problems, so let's just use setup.py so that we're at least doing what most projects do and using tools we all know. In the long run, pyproject.toml could potentially solve everything without ever using setuptools at all, but that remains uncharted territory and does not seem to be a practical short-term solution.
